### PR TITLE
Add ResolvedChannel

### PR DESCRIPTION
**Pull request by HopeBaron**
adds support to resolved channels by application commands

### CODE REVIEW

File: core/src/main/kotlin/entity/channel/ResolvedChannel.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+) : Channel {
+
+
+    val name: String get() = data.name.value!!
+
+    val permissions: Permissions get() = data.permissions.value!!
+
+    override val supplier: EntitySupplier
+        get() = strategy.supply(kord)
+
```
Comment (BartArys) [inline core/src/main/kotlin/entity/channel/ResolvedChannel.kt:24]: Should add a meaningful toString

**Review by BartArys — APPROVED** [review]

File: core/src/main/kotlin/entity/channel/ResolvedChannel.kt (hunk 1 above)
Comment (HopeBaron) [reply core/src/main/kotlin/entity/channel/ResolvedChannel.kt:24]: Should we add the object reference for the type, or it's value (Discord value) Although the object gives better idea of the actual type. However the value is less complicated than this: 
```kt
Type: dev.kord.common.entity.ChannelType$GuildCategory@47e3a62b
```

Comment (BartArys) [reply core/src/main/kotlin/entity/channel/ResolvedChannel.kt:24]: We should probably just give those types a better toString as well, though that's not the scope of this PR

File: core/src/main/kotlin/entity/channel/ResolvedChannel.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    override fun toString(): String {
+        return "ResolvedChannel(id=$id,type=$type, name=$name, permissions=$permissions)"
```
Comment (BartArys) [inline core/src/main/kotlin/entity/channel/ResolvedChannel.kt:26]: minor nitpick

```suggestion
        return "ResolvedChannel(id=$id, type=$type, name=$name, permissions=$permissions)"
```

**Review by BartArys — APPROVED** [review]

**Pull request merged**